src/services/layer-metadata/layer-metadata.service.ts

```diff
@@ -0,0 +1,155 @@
+import { ThemeNodeModel } from '@/composables/themes/themes.model'
+import { LayerTreeNodeModel } from '@/components/layer-tree/layer-tree.model'
+import { IdValues, LayerMetadataModel } from './layer-metadata.model'
+import {
+  getMetadataLinks,
+  getResponsibleParty,
+  isoLang2To3,
+  stringToHtml,
+} from './layer-metadata.utils'
+import { wmsHelper } from '../common/wms.helper'
+import useThemes from '../../composables/themes/themes.composable'
+
+export class LayerMetadataService {
+  // TODO: get urls from a config
+  private geonetworkBaseUrl =
+    'https://geocatalogue.geoportail.lu/geonetwork/srv'
+  private legendBaseUrl = 'https://map.geoportail.lu/legends/get_html'
+  // TODO: get from config or relative
+  private localMetadataBaseUrl = 'https://map.geoportail.lu/getMetadata'
+
+  async getLayerMetadata(
+    node: LayerTreeNodeModel,
+    currentLanguage: string
+  ): Promise<LayerMetadataModel> {
+    const layer: ThemeNodeModel | undefined = useThemes().findById(
+      parseInt(node.id, 10)
+    )
+    let localMetadata
+    let metadata
+    if (layer) {
```

Comment:
Did you copy all this code from old code base ? It looks like old fashion

Reply:
Yes, I started from the old code, should've rather started from scratch in the end, I think. I'll have a look at this method again when integrating the WMTS.